tests/installation/user_settings.pm

```diff
@@ -34,13 +34,18 @@ sub run {
         return;
     }
 
-    $self->enter_userinfo();
-    assert_screen 'inst-userinfostyped';
-    if (match_has_tag 'boo#1122804') {
-        diag "trying again with slow typing due to https://progress.opensuse.org/issues/46190\n";
-        $self->enter_userinfo(retry => 1);
-        assert_screen 'inst-userinfostyped';
-    }
+    # retry if not typed correctly
+    my $max_tries = 2;
+    my $retry     = 0;
+    do {
+        $self->enter_userinfo(retry => $retry);
+        assert_screen qw(inst-userinfostyped-ignore-full-name inst-userinfostyped-expected-typefaces);
+        if (!match_has_tag('inst-userinfostyped-expected-typefaces')) {
+            record_soft_failure('boo#1122804 - Typing issue with fullname');
+        }
+        $retry++;
+    } while (($retry < $max_tries) && !match_has_tag('inst-userinfostyped-expected-typefaces'));
```

Comment:
you could just do

```suggestion
my $retries = 2;
do {
    $self->enter_userinfo(retry => $retries);
    assert_screen qw(inst-userinfostyped-ignore-full-name inst-userinfostyped-expected-typefaces);
    last if match_has_tag('inst-userinfostyped-expected-typefaces');
    record_soft_failure 'boo#1122804 - Typing issue with fullname';
    --$retries;
} while ($retries > 0);
```

so no need for double if and no negation.

Reply:
@SergioAtSUSE wdyt?

Reply:
Ignoring the retries suggestion but I accept the "last if" to avoid multi condition

Reply:
how about

```suggestion
    record_soft_failure('boo#1122804 - Typing issue with fullname') unless match_has_tag('inst-userinfostyped-expected-typefaces');
```

at least :)

Regarding the two tags "inst-userinfostyped-ignore-full-name inst-userinfostyped-expected-typefaces". I don't think people creating the two needles would know how to create the needles correctly. I am also not sure what areas with what content would be needed.

Reply:
Nobody knows for sure what a needle should contain. The person that is going to create a needle should look at the previously created needle and the code (assert_and_click, check_screen, match_has_tag).

Reply:
And this is why I am against match_has_tag and why I had more than one assert_screen. My code was more reliably than this: http://openqa.slindomansilla-vm.qa.suse.de/tests/2168#step/user_settings/2

It is a matter of time to have a race condition where the button is pressed before the password confirmation has being typed.